addons/mail_group/security/mail_group_security.xml

```diff
@@ -0,0 +1,126 @@
+        <field name="domain_force">[
+            '&amp;',
+                ('moderation_status', '=', 'accepted'),
+                '|',
+                '|',
+                    ('mail_group_id.moderator_ids', 'in', user.id),
+                    ('mail_group_id.privacy', '=', 'public'),
+                    '&amp;',
+                        ('mail_group_id.privacy', '=', 'groups'),
+                        ('mail_group_id.group_public_id', 'in', [g.id for g in user.groups_id]),
+            ]
+        </field>
+        <field name="groups" eval="[(4, ref('base.group_public')), (4, ref('base.group_portal'))]"/>
+        <field name="perm_create" eval="True"/>
+        <field name="perm_read" eval="True"/>
+        <field name="perm_write" eval="True"/>
+        <field name="perm_unlink" eval="True"/>
+    </record>
+    <record id="mail_group_message_rule_user" model="ir.rule">
+        <field name="name">Mail Group Message: Non-accepted messages are accessible only by moderators</field>
+        <field name="model_id" ref="model_mail_group_message"/>
+        <field name="domain_force">[
+                '&amp;',
+                    '|',
+                        ('moderation_status', '=', 'accepted'),
+                        ('mail_group_id.moderator_ids', 'in', user.id),
+                    '|',
+                    '|',
+                        ('mail_group_id.moderator_ids', 'in', user.id),
+                        ('mail_group_id.privacy', '=', 'public'),
+                        '&amp;',
+                            ('mail_group_id.privacy', '=', 'groups'),
+                            ('mail_group_id.group_public_id', 'in', [g.id for g in user.groups_id]),
+            ]
+        </field>
+        <field name="groups" eval="[(4, ref('base.group_user'))]"/>
+        <field name="perm_create" eval="True"/>
+        <field name="perm_read" eval="True"/>
+        <field name="perm_write" eval="True"/>
+        <field name="perm_unlink" eval="True"/>
+    </record>
+    <record id="mail_group_message_rule_administrator" model="ir.rule">
+        <field name="name">Mail Group Message: Administrator have access to all messages</field>
+        <field name="model_id" ref="model_mail_group_message"/>
+        <field name="domain_force">[]</field>
+        <field name="groups" eval="[(4, ref('mail_group.group_mailing_list_administrator'))]"/>
+        <field name="perm_create" eval="True"/>
+        <field name="perm_read" eval="True"/>
+        <field name="perm_write" eval="True"/>
+        <field name="perm_unlink" eval="True"/>
+    </record>
+    <record id="mail_group_member_rule_user" model="ir.rule">
+        <field name="name">Mail Group Member: Members are accessible only by moderators</field>
+        <field name="model_id" ref="model_mail_group_member"/>
+        <field name="domain_force">[('mail_group_id.moderator_ids', 'in', user.id), ('mail_group_id.moderation', '=', True)]</field>
```

Comment:
Shouldn't we have a responsible on groups that manages members ? Otherwise on a non moderated group, nobody can manage members ... a bit strange.

Reply:
On non-moderated group, the "Mail Group Manager" is the only one who can manage them. I'm not sure it's a good idea to mix moderator / group manager / responsible, it become quickly complex

Reply:
That means you have to ping someone each time you want to do a modification on private groups ... group manager is someone that has cross-groups rights, but we need people to be able to create and configure their group.

Reply:
But so when a group is moderated, we will have 2 fields that do the same thing... :/